apps/ui/src/views/campaign/CampaignDetail.tsx

```diff
@@ -48,7 +48,7 @@ export default function CampaignDetail() {
     const [locale, setLocale] = useState<LocaleSelection>(localeState(templates ?? []))
     useEffect(() => {
```

Comment:
this part here seems like it should be a memo rather than effect + state. Also not sure we need a locale state separate from template since templates are already locale-specific? could just get it from there.